### ISSUE TITLE
Log Header is printed only once per file

### ISSUE THREAD

**Issue by ithr0n** [NLog/NLog#1713]
Hi!

When using newest NLog version, the header is printed only once when creating the log file.

In earlier version of NLog the behavior was:

Application starting -> print header
Application running -> print log entries
Application terminating -> print footer

Now it seems, that the header is printed only when the log file is created, but not when the log file already exists.

Did I miss something?

**Type**: Question
**NLog version**: 4.3.10
**Platform**: .Net 3.5 

**Current NLog config**

``` xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
      autoReload="true"
      throwExceptions="false"
      internalLogLevel="Off"
      internalLogFile="c:\temp\nlog-internal.log">

    <targets xsi:type="NLogTargets" async="false">

        <target xsi:type="File"
                name="fileTarget"
                fileName="${basedir}/logs/SvnBootstrapper.log"
                lineEnding="CRLF"
                archiveAboveSize="5048576"
                maxArchiveFiles="1"
                archiveNumbering="Rolling"
                header="${newline}
======================================================================================================================================================${newline}
Logging start = ${longdate}${newline}
Base dirctory = ${basedir}${newline}
Machine name  = ${machinename}${newline}
------------------------------------------------------------------------------------------------------------------------------------------------------${newline}
Date and Time            | Level | Module                                                | Message${newline}
------------------------------------------------------------------------------------------------------------------------------------------------------"
                layout=" ${longdate} | ${level:uppercase=true:padding=-5} | ${Logger:shortName=true:padding=20} | ${callsite:includeSourcePath=false:className=false:padding=30:fixedLength=true} | ${message} ${exception:format=message}"
                footer="${newline}
------------------------------------------------------------------------------------------------------------------------------------------------------${newline}
Logging end   = ${longdate}${newline}
======================================================================================================================================================" />
    </targets>

    <rules>

        <logger name="*" minlevel="Info" writeTo="fileTarget" />

    </rules>
</nlog>
```

**Comment by 304NotModified** [comment]
> Now it seems, that the header is printed only when the log file is created, but not when the log file already exists.

That's intended behavior. 

> In earlier version of NLog the behavior was:
> 
> Application starting -> print header
> Application running -> print log entries
> Application terminating -> print footer

That was a bug. 

Maybe we should make it configurable?

**Comment by 304NotModified** [comment]
I assume your question has been answered, if not, let us know!

**Comment by yarick123** [comment]
Would it be possible, to make it configurable?

**Comment by chuckleplant** [comment]
I'd also like the header to be printed for each session, not each log file. 

If my app crashes the footer isn't printed and it's hard to tell two sessions apart.

**Comment by 304NotModified** [comment]
What's a session in this case?

**Comment by chuckleplant** [comment]
@304NotModified , session would be an instance run of my application.  Starting and stopping it

**Comment by 304NotModified** [comment]
OK, please make a new issue so we could easier track this.

**Comment by snakefoot** [comment]
New option `WriteHeaderWhenInitialFileNotEmpty` has been introduced with NLog v5.2.6
